packages/make-styles/src/runtime/resolveStyleRules.ts

```diff
@@ -59,15 +82,16 @@ export function resolveStyleRules(
         support,
         value,
         unstable_cssPriority,
-
-        rtlProperty: flippedInRtl ? rtlDefinition.key : undefined,
-        rtlValue: flippedInRtl ? rtlDefinition.value : undefined,
+        ...rtlCompileOptions,
       });
-      const resolvedRule: MakeStylesResolvedRule = [getStyleBucketName(pseudo, media, support), className, ltrCSS];
 
-      if (rtlCSS) {
-        resolvedRule.push(rtlCSS);
-      }
+      const resolvedRule: MakeStylesResolvedRule = [
+        getStyleBucketName(pseudo, media, support),
+        className,
+        ltrCSS,
+        rtlCSS && rtlClassName,
```

Comment:
```suggestion
        rtlClassName,
```

Do we need a check there?

Reply:
Yeah, the idea is to only add `rtlClassName` if `rtlCSS` is also added, other wise this could be kind of confusing, to have a className that actually points to no actual style.